sdk/batch/microsoft-azure-batch/src/main/java/com/microsoft/azure/batch/BatchClient.java

```diff
@@ -98,7 +98,7 @@ public static BatchClient open(BatchCredentials credentials) {
      * @param baseUrl A  String object specifying the the batch end point.
      * @return The new {@link BatchClient} instance.
      */
-    static BatchClient open(RestClient restClient, String baseUrl) {
+    public static BatchClient open(RestClient restClient, String baseUrl) {
```

Comment:
I think if there's an issue in `FlatteningDeserializer`, then we should fix that issue instead of making an internal method public to work around this issue. Once this method is made public, it cannot be removed without causing a breaking change.

Reply:
Without this method, current users cannot make any changes to the proxy configuration or logging, or retry, or anything. The only public method to create a `BatchClient` allows passing in only the credential.

Reply:
Thanks for the clarification @jianghaolu . Ideally, we should have named this method `create()` but since there is already a public method called `open()` that creates a BatchClient using credentials, we should continue to use the same name.